scipy/interpolate/_rgi_cython.pyx

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+cimport numpy as np
+cimport cython
+
+np.import_array()
+
+import itertools
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+@cython.cdivision(True)
+cdef int find_interval_ascending(const double *x,
```

Comment:
Minor, can be done later: this function is straight from _ppoly.pyx, right? If it is a straight copy-paste, it would be nice to move this function to a  .pxi file and include in both places. Otherwise, please add comments on what changed wrt the _ppoly version.

Reply:
This is a straight copy from _ppoly.pyx. I'll add it into a .pxi file later

Reply:
FYI, [the use of `include` statement and files (`*.pxi`) is deprecated](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#the-include-statement-and-include-files). 
It is now possible to [share declarations between modules](https://cython.readthedocs.io/en/latest/src/userguide/sharing_declarations.html#sharing-declarations) instead.

On a side note, this function cannot be made more efficient.